Clean RPM deps

chouquette opened the pull request:
### What does this PR do?

Drop some optional RPM dependencies from our build

### Motivation

Stop building software we don't need. This will reduce a bit the agent disk footprint and also simplify the migration to bazel

### Describe how you validated your changes

### Additional Notes

chouquette committed:
omnibus: rpm: drop libdb dependency

chouquette committed:
omnibus: rpm: drop sqlite dependency

chouquette committed:
omnibus: rpm: drop elfutils dep

chouquette committed:
omnibus: rpm: drop ligcrypt dep

rdesgroppes commented on omnibus/config/software/rpm.rb:
```diff
@@ -22,14 +22,9 @@
 skip_transitive_dependency_licensing true
 
 dependency "config_guess"
-dependency "elfutils"
 dependency "file"
-dependency "libgpg-error"
-dependency "libgcrypt"
 dependency "popt"
 dependency "zstd"
-dependency "libsqlite3"
-dependency "libdb"
```
I'm curious how did you do manage to shrink it down? Looking at what's actually used downstream, relying on compilation/linking errors, comparing to `deb` deps, or something else?

chouquette replied:
Mostly looking at their build system & docs to see what we could reasonably disable
* elfutils is optional and only appears to be used when creating a package
* gcrypt/gpg-error shouldn't be needed when relying on the openssl crypto backend
* libsqlite appears to be overzealous after discussing with @0intro
* libdb seems unneeded, according to https://github.com/rpm-software-management/rpm/blob/master/INSTALL#L87C2-L89

alopezz replied:
Can we also get rid of `lua`?

rdesgroppes commented on omnibus/config/software/rpm.rb:
```diff
@@ -55,13 +50,12 @@
 
   configure_options = [
-    "--enable-sqlite=yes",
+    "--enable-ndb",
+    "--disable-sqlite",
```
Likewise

rdesgroppes reviewed (approved):
Thank you!

chouquette committed:
omnibus: rpm: build only the component we care for

this allows to drop more dependencies, at the cost of a fairly ugly
build command

chouquette committed:
also install rpmrc

chouquette committed:
Revert "omnibus: rpm: drop sqlite dependency"

This reverts commit 4b2a7362a9feb5efd3c9886a6111d1102873a684.

chouquette committed:
remove leftover comment

aiuto reviewed (approved):
Thanks for looking into this.
